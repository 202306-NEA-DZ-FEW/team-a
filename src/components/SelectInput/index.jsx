function SelectInput({
  data,
  name,
  label,
  handleChange,
  touched,
  error,
  value,
}) {
  return (
    <div>
      <div className='label'>
        <label htmlFor={name} className='label-text'>
          {label}
        </label>
        <label
          className={`label-text-alt ${touched && error ? "text-error" : ""}`}
        >
          {touched && error ? error : ""}
        </label>
      </div>
      <select
        id={name}
        className='select select-primary bg-white w-full'
        onChange={handleChange}
        defaultValue={value}
      >
        {data.map(({ name }) => (
          <option key={name} value={name}>
            {name}
          </option>
        ))}
      </select>
    </div>
  );
}

export default SelectInput;
