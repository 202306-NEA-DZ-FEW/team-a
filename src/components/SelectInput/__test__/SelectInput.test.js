import renderer from "react-test-renderer";

import SelectInput from "..";

it("renders correctly", () => {
  const mockData = [{ name: "foo" }, { name: "bar" }];
  const tree = renderer.create(<SelectInput data={mockData} />).toJSON();
  expect(tree).toMatchSnapshot();
});
