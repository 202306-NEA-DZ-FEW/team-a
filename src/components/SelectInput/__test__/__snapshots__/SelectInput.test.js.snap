// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="label"
  >
    <label
      className="label-text"
    />
    <label
      className="label-text-alt "
    />
  </div>
  <select
    className="select select-primary bg-white w-full"
  >
    <option
      value="foo"
    >
      foo
    </option>
    <option
      value="bar"
    >
      bar
    </option>
  </select>
</div>
`;
